tests/PlayerDaoTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+require_once 'PlayerDao.php';
+
+class PlayerDaoTest extends TestCase {
```

Comment:
On one hand, what you have done in this test is really nice, well done for figuring out how to do this.

On the other hand, you generally shouldn't test code that connects to a database, because the database may not always be available when the tests run (in the real world), and also because it's no longer a unit test. Unit tests are always testing one small part of code in isolation, these tests are testing both the Dao class and the database itself.

So while this is really good, you don't actually need this test file at all.

Reply:
Thank you for letting me know, much appreciated and amended!